### ISSUE TITLE
Items don't drop when combat logging on a Folia server

### ISSUE THREAD

**Issue by iLemon** [ChanceSD/PvPManager#509]
### Describe the bug
If a player is in combat and logs out on a Folia server, an error occurs and their items do not drop as intended.
Error: https://pastes.dev/K5jU69eabL
```
[13:43:50] [Region Scheduler Thread #0/INFO]: HOCCO2HCH2CO2H2 lost connection: Disconnected
[13:43:50] [Region Scheduler Thread #0/ERROR]: Could not pass event PlayerQuitEvent to PvPManager v3.33.3
java.lang.UnsupportedOperationException: null
	at org.bukkit.craftbukkit.v1_20_R3.scheduler.CraftScheduler.handle(CraftScheduler.java:535) ~[citrus-1.20.4.jar:git-Citrus-"36fda67"]
	at org.bukkit.craftbukkit.v1_20_R3.scheduler.CraftScheduler.runTaskTimer(CraftScheduler.java:235) ~[citrus-1.20.4.jar:git-Citrus-"36fda67"]
	at org.bukkit.craftbukkit.v1_20_R3.scheduler.CraftScheduler.runTaskTimer(CraftScheduler.java:211) ~[citrus-1.20.4.jar:git-Citrus-"36fda67"]
	at org.bukkit.scheduler.BukkitRunnable.runTaskTimer(BukkitRunnable.java:117) ~[citrus-api-1.20.4-R0.1-SNAPSHOT.jar:?]
	at nc.aL.<init>(SourceFile:26) ~[PvPManager-3.33.3.jar:?]
	at nc.aK.<init>(SourceFile:20) ~[PvPManager-3.33.3.jar:?]
	at nc.aJ.<init>(SourceFile:15) ~[PvPManager-3.33.3.jar:?]
	at nc.aN.a(SourceFile:1056) ~[PvPManager-3.33.3.jar:?]
	at nc.eW.a(SourceFile:51) ~[PvPManager-3.33.3.jar:?]
	at me.NoChance.PvPManager.Managers.PlayerHandler.b(SourceFile:310) ~[PvPManager-3.33.3.jar:?]
	at nc.eH.a(SourceFile:4294) ~[PvPManager-3.33.3.jar:?]
	at com.destroystokyo.paper.event.executor.MethodHandleEventExecutor.execute(MethodHandleEventExecutor.java:40) ~[citrus-api-1.20.4-R0.1-SNAPSHOT.jar:?]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:77) ~[citrus-api-1.20.4-R0.1-SNAPSHOT.jar:git-Citrus-"36fda67"]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[citrus-api-1.20.4-R0.1-SNAPSHOT.jar:?]
	at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54) ~[citrus-1.20.4.jar:git-Citrus-"36fda67"]
	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:126) ~[citrus-1.20.4.jar:git-Citrus-"36fda67"]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:617) ~[citrus-api-1.20.4-R0.1-SNAPSHOT.jar:?]
	at net.minecraft.server.players.PlayerList.remove(PlayerList.java:711) ~[citrus-1.20.4.jar:git-Citrus-"36fda67"]
	at net.minecraft.server.players.PlayerList.remove(PlayerList.java:696) ~[citrus-1.20.4.jar:git-Citrus-"36fda67"]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.removePlayerFromWorld(ServerGamePacketListenerImpl.java:2144) ~[?:?]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.onDisconnect(ServerGamePacketListenerImpl.java:2122) ~[?:?]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.onDisconnect(ServerGamePacketListenerImpl.java:2109) ~[?:?]
	at net.minecraft.network.Connection.handleDisconnection(Connection.java:895) ~[?:?]
	at io.papermc.paper.threadedregions.RegionizedWorldData.tickConnections(RegionizedWorldData.java:516) ~[citrus-1.20.4.jar:git-Citrus-"36fda67"]
	at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1845) ~[citrus-1.20.4.jar:git-Citrus-"36fda67"]
	at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:449) ~[citrus-1.20.4.jar:git-Citrus-"36fda67"]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1659) ~[citrus-1.20.4.jar:git-Citrus-"36fda67"]
	at io.papermc.paper.threadedregions.TickRegions$ConcreteRegionTickHandle.tickRegion(TickRegions.java:417) ~[citrus-1.20.4.jar:git-Citrus-"36fda67"]
	at io.papermc.paper.threadedregions.TickRegionScheduler$RegionScheduleHandle.runTick(TickRegionScheduler.java:404) ~[citrus-1.20.4.jar:git-Citrus-"36fda67"]
	at ca.spottedleaf.concurrentutil.scheduler.SchedulerThreadPool$TickThreadRunner.run(SchedulerThreadPool.java:525) ~[citrus-1.20.4.jar:git-Citrus-"36fda67"]
	at java.lang.Thread.run(Thread.java:1583) ~[?:?]
```

### To reproduce the bug

3.33.3, and 1.20.4 Folia

**Comment by ChanceSD** [comment]
Thanks for the report, shouldn't be too hard to fix, will test it soon.

**Comment by ChanceSD** [comment]
Should be resolved in v3.34.
